cmd/soroban-rpc/internal/methods/get_transactions.go

```diff
@@ -0,0 +1,230 @@
+package methods
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	"github.com/creachadair/jrpc2"
+	"github.com/creachadair/jrpc2/handler"
+	"github.com/stellar/go/ingest"
+	"github.com/stellar/go/support/errors"
+	"github.com/stellar/go/support/log"
+	"github.com/stellar/go/toid"
+
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/db"
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/ledgerbucketwindow"
+)
+
+// TransactionsPaginationOptions defines the available options for paginating through transactions.
+type TransactionsPaginationOptions struct {
+	Cursor *toid.ID `json:"cursor,omitempty"`
```

Comment:
the cursor type in the request does not match the cursor type in GetTransactionsResponse. The cursor you provide in the request should probably be a string and not a json object

Reply:
@tamirms I used the json object because that is how we use it for getEvents pagination options. If we change this cursor to string, we should also do it for the events cursor to keep it same. What do you think?

Reply:
@aditya1702 the curser for getEvents is a string.  https://developers.stellar.org/network/soroban-rpc/api-reference/methods/getEvents

the events cursor struct has a custom json implementation which serializes the struct into a string

https://github.com/stellar/soroban-rpc/blob/main/cmd/soroban-rpc/internal/events/cursor.go#L39-L57

Reply:
@tamirms Gotcha. So in this case I would have to explicitly parse the cursor string using toid.Parse right? As I have done in this test - https://github.com/stellar/soroban-rpc/pull/136/files#diff-dd051ad5301911cdd3964f2129a8d3e93066ca9bd9a9490166fdab6cf28d3952R92

Reply:
yeah, you can make it a string and parse the value into a toid.ID or you could add a wrapper around toid.ID which defines json marshaling / unmarshaling functions

Reply:
I will just parse it as string and convert to toid.ID